ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/PreExecutingValidatingCommitter.scala

```diff
@@ -68,9 +72,8 @@ class PreExecutingValidatingCommitter[LogResult](
               CachingDamlLedgerStateReaderWithFingerprints(
                 stateValueCache,
                 cacheUpdatePolicy,
-                new LedgerStateReaderWithFingerprintsFromValues(
-                  ledgerStateOperations,
-                  valueToFingerprint),
+                new LedgerStateOperationsReader(ledgerStateOperations)
+                  .mapValues(value => value -> valueToFingerprint(value)),
```

Comment:
Nice solution!

Reply:
More type parameters pushed me in this direction.